src/utils/templates.ts

```diff
@@ -85,21 +98,18 @@ function copyRemote(sourcePath: URL, targetFolder: string, fileName: string) {
   });
 }
 
-async function copyLocal(sourcePath: string, targetFolder: string, fileName: string, options: { [key: string]: any }) {
-  let content: string;
+async function copyLocal(sourcePath: string, fileName: string, options: { [key: string]: any }) {
+  fs.mkdirSync(path.dirname(fileName), { recursive: true });
 
   if (path.extname(sourcePath) === '.mu') {
     const template = compile(fs.readFileSync(sourcePath, { encoding: 'utf8' }));
-    content = template(options);
+    fs.writeFileSync(fileName, template(options));
   } else {
-    content = fs.readFileSync(sourcePath, { encoding: 'utf8' });
+    fs.copyFileSync(sourcePath, fileName);
   }
-
-  fs.mkdirSync(targetFolder, { recursive: true });
-  fs.writeFileSync(fileName, content);
 }
 
-export function ensureDirectoryExistence(filePath: string, isDir: boolean = false) {
+export function ensureDirectoryExists(filePath: string, isDir: boolean = false) {
```

Comment:
I guess we could replace this function with `fs.mkdirSync(dirname, { recursive: true });` everywhere, since you are already touching it.

Reply:
Are you sure? I don't really dare doing that as it will fail in case a directory already exists.

Reply:
> Calling fs.mkdir() when path is a directory that exists results in an error only when recursive is false.

From the node docs. So this should be safe even if the dir exists.

Reply:
uuh, ok 😊